cmd/qat_plugin/README.md

```diff
@@ -1,39 +1,215 @@
+table summarises the defaults:
+
+| Argument | Variable | Default setting | Explanation |
+|:-------- |:-------- |:--------------- |:----------- |
+| -dpdk-driver | `$DPDK_DRIVER` | vfio-pci | A more robust and secure choice than the `igb_uio` alternative |
+| -kernel-vf-drivers | `$KERNEL_VF_DRIVERS` | dh895xccvf,c6xxvf,c3xxxvf,d15xxvf | Modify to suit your hardware setup |
+| -max-num-devices | `$MAX_NUM_DEVICES` | 32 | Modify to suit your hardware setup if necessary |
+| -debug | `$DEBUG` | false | Debug is disabled by default |
+
+For more details on the `-dpdk-driver` choice, see 
+[DPDK Linux Driver Guide](http://dpdk.org/doc/guides/linux_gsg/linux_drivers.html).
+
+> **FIXME** - I could not find any information around `kernel-vf-drivers` on the following site:
+> 
+> Use the `kernel-vf-drivers flag` to specify the vf Device Driver for the particular QAT device. For more information, refer to [Intel QAT Crypto Poll Mode Driver](https://dpdk.org/doc/guides/cryptodevs/qat.html).
+
+If the `-mode` parameter is set to `kerneldrv`, no other parameter documented above are valid,
+except `-debug` which is global for both modes.
+`kerneldrv` mode implements resource allocation based on system configured [logical instances][7].
+
+The `kerneldrv` mode does not guarantee full device isolation between containers
+and therefore it's not recommended. This mode will be deprecated and removed once `libqat`
+implements non-UIO based device access.
+
+# Installation
+
+The below sections cover how to obtain, build and install this component.
+
+The component can be installed either using a DaemonSet or running 'by hand' on each node.
+
+## Pre-built image
+
+[Pre-built images](https://hub.docker.com/r/intel/intel-qat-plugin)
+of this component are available on the Docker hub. These images are automatically built and uploaded
+from the latest master branch of this repository.
+
+Release tagged images of the components are also available on the Docker hub, tagged with their
+release version numbers.
+
+The deployment YAML files supplied with this component in this repository use the images with the `devel`
+tag by default. If you do not build your own local images, your Kubernetes cluster may pull down
+the devel images from the Docker hub by default.
+
+To use the release tagged versions of the images, edit the
+[YAML deployment files](../../deployments/qat_plugin/)
+appropriately.
+
+## Prerequisites
+
+The component has the same basic dependancies as the
+[generic plugin framework dependencies](../../README.md#about).
+
+You will also need [appropriate hardware installed](#checking-for-hardware).
+
+The Intel QuickAssist Technology software for Linux must be installed and configured.
+For more information, refer to the
+[Intel QuickAssist Technology Software for Linux - Getting Started Guide](https://01.org/sites/default/files/downloads/intelr-quickassist-technology/336212qatswgettingstartedguiderev003.pdf)
```

Comment:
based on what I wrote earlier about the upstream/in-kernel drivers, I believe this could actually be removed/changed to a list of `CONFIG_` entries. Thoughts?

Reply:
I guess we need to say there are two kernel drivers - in and out of tree - and you need at least one of them installed/enabled - and provide links to the two.
We should probably note any differences or limitations they each have as well.

I've updated/reworded to link to the two options. I've not added details on differences and limitations, as I don't know what they are ;-) Free free to add verbage here and I can merge that into this PR.

Reply:
Oh, and, yes, if setting the kernel `CONFIG` options is critical and tricky, it might be an idea to document here (or, we could point to the Kata docs - but, that somehow feels a little backwards...)

Reply:
the `CONFIG` options are business as usual but distros support it differently. Ubuntu does not have the modules at all: https://bugs.launchpad.net/ubuntu/+source/linux/+bug/1845959, Clear Linux has them 'built-in' which triggers problems with the firmware blobs (one needs to install `linux-firmware-qat` bundle for that)